### ISSUE TITLE
[Gen2 CF-HPG v1.2] Test deeper convolution-free raw-patch tokenizer after v1.1 partial resolution signal

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#48]
## Objective

Implement **CF-HPG v1.2 Tokenizer** as exactly one diagnosis-driven architecture change after the independently accepted Issue #47 outcome `RESOLUTION_PARTIAL_SIGNAL`.

Primary hypothesis:

> Increasing spatial resolution from 64 to 144 generic patch nodes produced a meaningful +4.51 pp validation-accuracy gain and +4.03 pp clean-train-accuracy gain, but absolute fitting remains weak (60.54% clean train, 57.34% validation). The remaining bottleneck may therefore be the shallow local visual tokenizer: each 4x4 raw patch currently receives only one linear `16 -> 96` projection before positional fusion and graph reasoning. Adding one shared nonlinear local refinement layer should increase patch-level visual expressivity without introducing convolution, anatomy, handcrafted features, attention, or additional graph depth.

This issue is **implementation + bounded synthetic verification only**. It does not authorize a FER2013 scientific training run. Execution must be separately preregistered after independent PR review.

## Locked evidence motivating this issue

Accepted v1.1 Issue #47 seed42 result:

- source/merge commit: `934057f60a6cb3bc7ec247e000573a40c390a426`
- model SHA-256: `87723409fd6f22d333b9bded2b8707d4e4a3bd5bd3bac19e48022f12f6df0fea`
- params: `411527`
- selected epoch: zero-based `77`
- clean-train accuracy: `0.605385070883695`
- clean-train macro-F1: `0.5474748347135944`
- validation accuracy: `0.5734187796043466`
- validation macro-F1: `0.5154247791468033`
- accuracy gap: `3.196629127934836 pp`
- macro-F1 gap: `3.205005556679108 pp`
- accepted decision: `RESOLUTION_PARTIAL_SIGNAL`
- evidence ZIP SHA-256: `4798078eee23d34e36d44f25d2fc977adf55452d0ec2279946149e80bf8dfe8c`

Accepted deltas vs v1.0:

- validation accuracy `+4.513792142658124 pp`
- validation macro-F1 `+5.272896090308943 pp`
- clean-train accuracy `+4.033578320387332 pp`
- clean-train macro-F1 `+4.445204648958034 pp`

The resolution hypothesis therefore has a real partial signal, but v1.1 remains practically underfit. Do not continue an opportunistic resolution sweep in this issue.

## Exact implementation base

Branch from exact current main:

`934057f60a6cb3bc7ec247e000573a40c390a426`

Do not modify existing reviewed candidates:

- `research/candidates/tf_cf_hpg/`
- `research/candidates/tf_cf_hpg_v1_1_resolution/`

Create isolated candidate, preferred:

`research/candidates/tf_cf_hpg_v1_2_tokenizer/`

Generation-1 and Step-13 sources remain read-only.

## Exactly one conceptual architecture change

Keep the v1.1 4x4 / 144-node graph and all graph/hierarchy/training semantics frozen.

### v1.1 raw-patch path

```text
raw 4x4 patch (16)
 -> Dense 16->96
 -> add positional projection 2->96
 -> LayerNorm -> GELU -> Dropout(0.10)
```

### v1.2 Tokenizer raw-patch path

```text
raw 4x4 patch (16)
 -> Dense 16->96                 [same raw projection]
 -> GELU                         [new local nonlinearity]
 -> Dense 96->96                 [new shared raw-local refinement]
 -> add positional projection 2->96
 -> LayerNorm -> GELU -> Dropout(0.10)
```

Only the new raw-local `GELU -> Dense(96)` refinement is added before positional fusion.

No residual tokenizer branch, no extra normalization, no convolution, no attention, no overlapping patches, no multiscale branch.

Expected exact parameter count:

`420839`

Reason: v1.1 `411527` + one new Dense `96->96` = `96*96 + 96 = 9312` parameters.

Expected trainable variable count: `66`.
Expected total Keras variable count: `76`.

If exact implementation does not produce these identities, STOP and report; do not alter widths to force a count.

## Everything else is frozen from v1.1

- input `48x48x1`
- scaling `x/127.5 - 1`
- non-overlapping patch `4x4`
- fine grid `12x12 = 144`
- raw patch width `16`
- generic normalized `(x,y)` coordinates only
- positional projection `2->96`
- fine width `96`
- exactly 2 fine Max-Relative blocks
- fine hybrid graph = spatial 8-neighborhood UNION cosine top-4 relations
- no self edges; exact deduplication
- fine graph built once and reused for both fine blocks
- exact parameter-free non-overlapping 2x2 arithmetic-mean hierarchy `12x12 -> 6x6`
- coarse nodes `36`
- coarse projection `96->128`
- coarse width `128`
- exactly 2 coarse Max-Relative blocks
- coarse spatial-8 UNION cosine-top-4 graph
- coarse graph built once and reused for both coarse blocks
- global mean + global max readout
- LayerNorm -> Dense `256->128` -> GELU -> Dropout(0.20) -> Dense `128->7`
- no model softmax

## Non-negotiable scientific boundary

Forbidden:

- Conv2D / DepthwiseConv2D / SeparableConv2D
- any convolutional patch tokenizer
- pretrained visual backbone/features
- MediaPipe / landmarks / AUs
- ROI / RoIAlign
- anatomical masks / part_soft / semantic channels
- HOG/LBP/Gabor/gradient/Laplacian handcrafted channels
- attention layers
- learned semantic/anatomical slots
- motif/query banks
- patch-resolution change
- overlapping patches
- graph top-k change
- graph block count change
- fine/coarse width change
- pooling/hierarchy change
- readout change
- augmentation or optimizer change

This is a **local-tokenizer-depth-only falsification**.

## Future training lifecycle to prepare, NOT run here

Copy v1.1 lifecycle exactly, changing only candidate/model/output identity and diagnostic reference.

Frozen future config:

- seed `42`
- train `28709`, validation `3589`, test forbidden
- AdamW
- LR `3e-4`
- weight decay `5e-4`
- global clipnorm `1.0`
- batch `64`
- max epochs `100`
- warmup `5`
- cosine to `1e-6`
- label smoothing `0.05`
- earliest strict maximum `val_accuracy` checkpoint
- early stopping on `val_loss`, patience `15`, min_delta `0`
- same train-only augmentation and exact deterministic semantics as v1.1
- validation and clean-train augmentation OFF
- same fail-closed test-path isolation

No full FER run in this issue.

## Preregistered future comparison reference

Use accepted v1.1 only:

- clean-train accuracy `0.605385070883695`
- clean-train macro-F1 `0.5474748347135944`
- validation accuracy `0.5734187796043466`
- validation macro-F1 `0.5154247791468033`

Future execution must report:

- `delta_val_accuracy_pp = 100*(v1.2_val_acc - 0.5734187796043466)`
- `delta_val_macro_pp = 100*(v1.2_val_macro - 0.5154247791468033)`
- `delta_clean_train_accuracy_pp = 100*(v1.2_clean_train_acc - 0.605385070883695)`
- `delta_clean_train_macro_pp = 100*(v1.2_clean_train_macro - 0.5474748347135944)`

## Future decision precedence

Freeze before outcome:

1. `CF_HPG_V1_2_STRETCH_PASS`
2. `CF_HPG_V1_2_PASS`
3. `TOKENIZER_OVERFIT_SHIFT`
4. `TOKENIZER_STRONG_SIGNAL`
5. `TOKENIZER_PARTIAL_SIGNAL`
6. `TOKENIZER_UNDERFIT_REMAINS`
7. `TOKENIZER_INCONCLUSIVE`

### `CF_HPG_V1_2_STRETCH_PASS`
All:
- val accuracy `>=0.7000`
- val macro-F1 `>=0.6700`
- both clean gaps `<=8 pp`

### `CF_HPG_V1_2_PASS`
All:
- val accuracy `>=0.6500`
- val macro-F1 `>=0.6200`
- both clean gaps `<=8 pp`

### `TOKENIZER_OVERFIT_SHIFT`
- clean-train accuracy improvement vs v1.1 `>=5.0 pp`
- and either clean gap `>10 pp`

### `TOKENIZER_STRONG_SIGNAL`
- validation accuracy improvement vs v1.1 `>=5.0 pp`
- clean-train accuracy improvement vs v1.1 `>=5.0 pp`
- only if no higher-priority label applies

### `TOKENIZER_PARTIAL_SIGNAL`
- validation accuracy improvement vs v1.1 `>=3.0 pp` and `<5.0 pp`
- only if no higher-priority label applies

### `TOKENIZER_UNDERFIT_REMAINS`
- clean-train accuracy `<0.7000`
- validation accuracy `<0.6400`
- only if no higher-priority label applies

Else `TOKENIZER_INCONCLUSIVE`.

These are single-seed diagnostic labels, not statistical superiority/equivalence claims.

## Required implementation tests

Synthetic/tiny only. At minimum prove:

1. v1.1 patchification/grid/hierarchy shapes remain exact;
2. exact raw tokenizer order `Dense16->96 -> GELU -> Dense96->96` before positional addition;
3. positional projection remains separate and is added only after raw-local refinement;
4. exactly one new Dense layer relative to v1.1 model path;
5. exact params `420839`;
6. exact trainable variables `66`, total variables `76`;
7. no convolution/attention/semantic/prior imports;
8. fine/coarse graph construction and reuse unchanged;
9. exact Max-Relative golden behavior unchanged;
10. exact 2x2 mean hierarchy unchanged;
11. readout unchanged;
12. data/augmentation source byte-identical to v1.1 where copied;
13. graph source byte-identical to v1.1 where copied;
14. exact future optimizer/scheduler/checkpoint/early-stop contract;
15. exact test-path fail-closed behavior, including validation-sized 3589 forbidden-path cases;
16. synthetic `tf.function` forward/backward finite and optimizer updates;
17. Keras save/load exact synthetic-logit round-trip;
18. existing v1.0 and v1.1 candidates unchanged;
19. Generation-1/Step-13 diff empty;
20. `git diff --check` PASS.

Also add exact decision-boundary and overlap-precedence tests for the frozen future labels above.

## PR boundary

Create a new branch from exact `934057f60a6cb3bc7ec247e000573a40c390a426` and a separate DRAFT PR linked to this issue.

Do not modify or merge earlier candidates.
Do not run FER.
Do not access test.
Do not launch Kaggle scientific execution.
Do not start another refinement.

Evidence status:

`CF_HPG_V1_2_TOKENIZER_IMPLEMENTATION_ONLY`
